pkg/argocd/git.go

```diff
@@ -128,6 +129,36 @@ func TemplateBranchName(branchName string, changeList []ChangeEntry) string {
 
 type changeWriter func(app *v1alpha1.Application, wbc *WriteBackConfig, gitC git.Client) (err error, skip bool)
 
+// getWriteBackBranch returns the branch to use for write-back operations.
+// It first checks for a branch specified in annotations, then uses the
+// targetRevision from the matching git source, falling back to getApplicationSource.
+func getWriteBackBranch(app *v1alpha1.Application) string {
+	// First check if branch is explicitly set in annotations
+	annotations := app.GetAnnotations()
+	if branch, ok := annotations[common.GitBranchAnnotation]; ok {
+		return branch
+	}
```

Comment:
we cannot take the annotation value as is, since it can be a composite form like "main:changes". See https://argocd-image-updater.readthedocs.io/en/latest/basics/update-methods/#specifying-a-branch-to-commit-to

I guess we dont' see the above block. Line 200 already handles this case. See next comment.